tests/test_minimal/test_tools/test_backend_and_dataset_configuration/test_helpers/test_get_default_dataset_io_configurations.py

```diff
@@ -0,0 +1,194 @@
+import pytest
+from hdmf.common import VectorData
+from hdmf.data_utils import DataChunkIterator
+from nwbinspector.utils import is_module_installed
+from pynwb.base import DynamicTable
+from pynwb.behavior import CompassDirection
+from pynwb.image import ImageSeries
+from pynwb.testing.mock.base import mock_TimeSeries
+from pynwb.testing.mock.behavior import mock_SpatialSeries
+from pynwb.testing.mock.file import mock_NWBFile
+
+from neuroconv.tools.hdmf import SliceableDataChunkIterator
+from neuroconv.tools.nwb_helpers import (
+    DATASET_IO_CONFIGURATIONS,
+    get_default_dataset_io_configurations,
+    get_module,
+)
+
+
+@pytest.mark.parametrize("iterator", [lambda x: x, SliceableDataChunkIterator, DataChunkIterator])
+@pytest.mark.parametrize("backend", ["hdf5", "zarr"])
+def test_configuration_on_time_series(iterator: callable, backend: Literal["hdf5", "zarr"]):
+    array = np.array([[1, 2, 3], [4, 5, 6]])
+    data = iterator(array)
+
+    nwbfile = mock_NWBFile()
+    time_series = mock_TimeSeries(name="TestTimeSeries", data=data)
+    nwbfile.add_acquisition(time_series)
+
+    dataset_configurations = list(get_default_dataset_io_configurations(nwbfile=nwbfile, backend=backend))
+
+    assert len(dataset_configurations) == 1
+
+    dataset_configuration = dataset_configurations[0]
+    assert isinstance(dataset_configuration, DATASET_IO_CONFIGURATIONS[backend])
+    assert dataset_configuration.dataset_info.object_id == time_series.object_id
+    assert dataset_configuration.dataset_info.location == "acquisition/TestTimeSeries/data"
+    assert dataset_configuration.dataset_info.full_shape == array.shape
+    assert dataset_configuration.dataset_info.dtype == array.dtype
+    assert dataset_configuration.chunk_shape == array.shape
+    assert dataset_configuration.buffer_shape == array.shape
+    assert dataset_configuration.compression_method == "gzip"
+    assert dataset_configuration.compression_options is None
+
+    if backend == "zarr":
+        assert dataset_configuration.filter_methods is None
+        assert dataset_configuration.filter_options is None
+
+
+@pytest.mark.parametrize("backend", ["hdf5", "zarr"])
+def test_configuration_on_external_image_series(backend: Literal["hdf5", "zarr"]):
+    nwbfile = mock_NWBFile()
+    image_series = ImageSeries(name="TestImageSeries", external_file=[""], rate=1.0)
+    nwbfile.add_acquisition(image_series)
+
+    dataset_configurations = list(get_default_dataset_io_configurations(nwbfile=nwbfile, backend=backend))
+
+    assert len(dataset_configurations) == 0
+
+
+@pytest.mark.parametrize("iterator", [lambda x: x, SliceableDataChunkIterator, DataChunkIterator])
+@pytest.mark.parametrize("backend", ["hdf5", "zarr"])
+def test_configuration_on_dynamic_table(iterator: callable, backend: Literal["hdf5", "zarr"]):
+    array = np.array([0.1, 0.2, 0.3])
+    data = iterator(array)
+
+    nwbfile = mock_NWBFile()
+    column = VectorData(name="TestColumn", description="", data=data)
```

Comment:
Can we add a ragged array column in the test? I am wondering how it wil work with the shape and dtype

Reply:
Very good thinking on this request. Caught a bug where `_index` columns weren't found

And in general, a good reminder for myself not to access column fields of a table by accessing their name as a key in the dictionary - since they override `__get_item__` for user convenience it actually has different behavior than I was expecting from a simple mapping